### ISSUE TITLE
scrollspy problem with two navs

### ISSUE THREAD

**Issue by devnieL** [twbs/bootstrap#15882]
Hi, it seems that scrollspy goes crazy with two navs targeting same elements, I have this approach because of responsive design, one navigation menu is hidden in certain dimensions.

**Demo** : http://jsbin.com/xowuw/1/

By now, my workaround is a simple filter in scrollspy.js' refresh method : 

```javascript
       var exists = false;

        for(var i in self.targets){
          console.log(self.targets[i], " == ", this[1]);

          if(self.targets[i] == this[1]){
            exists = true;
            break;
          }
        }

        if(!exists){
          self.offsets.push(this[0])
          self.targets.push(this[1])
        }
```

Thanks !

**Comment by twbs-lmvtfy** [comment]
Hi @devnieL!

You appear to have posted a live example (http://jsbin.com/xowuw/1/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 39, column 17: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 65, column 31: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 77, column 31: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 93, column 31: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 103, column 31: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 50, column 21: [E029](https://github.com/twbs/bootlint/wiki/E029): Since grid classes apply to devices with screen widths greater than or equal to the breakpoint sizes (unless overridden by grid classes targeting larger screens), `class="xxx-modal-scroll-menu-wrapper col-md-4 col-sm-3 col-xs-3 hidden-xs"` is redundant and can be simplified to `class="xxx-modal-scroll-menu-wrapper hidden-xs col-xs-3 col-md-4"`
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by devnieL** [comment]
Hi, it's fixed and the error still persists: 

http://jsbin.com/xowuw/2/ 

Thanks

**Comment by hnrch02** [comment]
I'm inclined to say this is a WontFix as it isn't the intended usage. @cvrebert What do you say?

**Comment by cvrebert** [comment]
Duplicate of #13071 perhaps?

**Comment by hnrch02** [comment]
Was thinking about that one too but here it's about two separate navs and not single elements in one nav. Might not make a difference though.

**Comment by cvrebert** [comment]
I am sympathetic to that argument, although the fact that [the docs example](http://getbootstrap.com/javascript/#scrollspy-examples) currently uses `data-spy="scroll" data-target=".navbar-example"` (a class selector rather than an ID selector) certainly isn't helping things.
